File: sdk/fsharp/src/HiTechCloudSDK/Collocation.fs
namespace HiTechCloud.SDK

/// Collocation Services: PDU port management
type CollocationResource(httpClient: HttpClient) =

    /// GET /api/service/{id}/pdu - List PDU ports assigned to service
    member this.ListPduPorts() =
        async {
            return! httpClient.getAsync("/api/service/{id}/pdu")
        }

    /// GET /api/service/{id}/pdu/{port} - Get PDU port/outlet state
    member this.GetPduPortState() =
        async {
            return! httpClient.getAsync("/api/service/{id}/pdu/{port}")
        }

    /// POST /api/service/{id}/pdu/{port} - Set PDU port/outlet state
    member this.SetPduPortStatus() =
        async {
            return! httpClient.postAsync("/api/service/{id}/pdu/{port}")
        }

File: sdk/clojure/src/hitechcloud/sdk/collocation.clj
(ns hitechcloud.sdk.collocation
  (:require [hitechcloud.sdk.http-client :as http]))

;; Collocation Services: PDU port management

(defn list_pdu_ports "GET /api/service/{id}/pdu - List PDU ports assigned to service" [client & [params]] (http-get client "/api/service/{id}/pdu" params))

(defn get_pdu_port_state "GET /api/service/{id}/pdu/{port} - Get PDU port/outlet state" [client & [params]] (http-get client "/api/service/{id}/pdu/{port}" params))

(defn set_pdu_port_status "POST /api/service/{id}/pdu/{port} - Set PDU port/outlet state" [client & [params]] (http-post client "/api/service/{id}/pdu/{port}" params))

